app/scripts/lib/util.ts

```diff
@@ -480,3 +481,97 @@ export function getConversionRatesForNativeAsset({
 
   return conversionRateResult;
 }
+
+// Cache for known domains
+let knownDomainsSet: Set<string> | null = null;
+let initPromise: Promise<void> | null = null;
+
+/**
+ * Initialize the set of known domains from the chains list
+ */
+export async function initializeRpcProviderDomains(): Promise<void> {
+  if (initPromise) {
+    return initPromise;
+  }
+
+  initPromise = (async () => {
+    try {
+      const chainsList = await getSafeChainsList();
+      knownDomainsSet = new Set<string>();
+
+      for (const chain of chainsList) {
+        if (chain.rpc && Array.isArray(chain.rpc)) {
+          for (const rpcUrl of chain.rpc) {
+            try {
+              // Removed cleaning step - will fail for URLs like:
+              // https://${INFURA_API_KEY}.infura.io/v3/
```

Comment:
This comment looks like it might be a reply to my previous PR review comment? I don't think it should be in the code.

That said,
`new URL("https://{INFURA_API_KEY}.infura.io/v3/").hostname` returns `${infura_api_key}.infura.io`. This seems like _correct_ behavior to me.